include/ostatus.php

```diff
@@ -396,973 +410,631 @@ public static function import($xml,$importer,&$contact, &$hub) {
-					return false;
+	/**
+	 * @brief Processes links in the XML
+	 *
+	 * @param object $links The xml data that contain links
+	 * @param array $item The item array
+	 *
+	 * @return array with data from the links
+	 */
+	private static function processLinks($links, &$item) {
+		$link_data = array('add_body' => '', 'self' => '');
+
+		foreach ($links AS $link) {
+			$attribute = self::read_attributes($link);
+
+			if (($attribute['rel'] != "") && ($attribute['href'] != "")) {
+				switch ($attribute['rel']) {
+					case "alternate":
+						$item["plink"] = $attribute['href'];
+						if (($item["object-type"] == ACTIVITY_OBJ_QUESTION) ||
+							($item["object-type"] == ACTIVITY_OBJ_EVENT)) {
+							$item["body"] .= add_page_info($attribute['href']);
+						}
+						break;
+					case "ostatus:conversation":
+						$link_data['conversation'] = $attribute['href'];
+						$item['conversation-href'] = $link_data['conversation'];
+						if (!isset($item['conversation-uri'])) {
+							$item['conversation-uri'] = $item['conversation-href'];
+						}
+						break;
+					case "enclosure":
+						$filetype = strtolower(substr($attribute['type'], 0, strpos($attribute['type'],'/')));
+						if ($filetype == 'image') {
+							$link_data['add_body'] .= "\n[img]".$attribute['href'].'[/img]';
+						} else {
+							if (strlen($item["attach"])) {
+								$item["attach"] .= ',';
+							}
+							if (!isset($attribute['length'])) {
+								$attribute['length'] = "0";
+							}
+							$item["attach"] .= '[attach]href="'.$attribute['href'].'" length="'.$attribute['length'].'" type="'.$attribute['type'].'" title="'.$attribute['title'].'"[/attach]';
+						}
+						break;
+					case "related":
+						if ($item["object-type"] != ACTIVITY_OBJ_BOOKMARK) {
+							if (!isset($item["parent-uri"])) {
+								$item["parent-uri"] = $attribute['href'];
+							}
+							$link_data['related'] = $attribute['href'];
+						} else {
+							$item["body"] .= add_page_info($attribute['href']);
+						}
+						break;
+					case "self":
+						if ($item["plink"] == '') {
+							$item["plink"] = $attribute['href'];
+						}
+						$link_data['self'] = $attribute['href'];
+						break;
 				}
 			}
-
-			$item_stored = item_store($item, $all_threads);
-			if ($item_stored) {
-				logger("Uri ".$item["uri"]." wasn't found in conversation ".$conversation_url, LOGGER_DEBUG);
-				self::store_conversation($item_stored, $conversation_url);
-			}
 		}
-
-		return $item_stored;
+		return $link_data;
 	}
 
-	/**
-	 * @brief Stores conversation data into the database
+/**
+	 * @brief Create an url out of an uri
 	 *
-	 * @param integer $itemid The id of the item
-	 * @param string $conversation_url The uri of the conversation
+	 * @param string $href URI in the format "parameter1:parameter1:..."
+	 *
+	 * @return string URL in the format http(s)://....
 	 */
-	private static function store_conversation($itemid, $conversation_url) {
+	public static function convert_href($href) {
```

Comment:
I would rather use camelCase for function naming (`public static function convertHref($href)`)

Reply:
This is an old function. I hadn't rewritten every code line.